docs/enhancements/image-handling-service.md

```diff
@@ -0,0 +1,171 @@
+
+#### Story 2
+
+An additional deployment and service will be managed for all deployment methods.
+This includes SaaS, operator, and CI (minikube). Whether an additional image is
+created is still an open question.
+
+### Implementation Details/Notes/Constraints
+
+#### Authentication and Authorization
+
+The image service will not need to implement authentication or authorization
+directly as it doesn't manage any user data. The user will provide a token to
+the image service and the image service will directly pass that token to
+assisted service on each call. Assisted service will validate the token before
+giving the image service any user information.
+
+#### Template Images
+
+Currently, when the assisted service starts, it downloads and caches each RHCOS
+image specified in the `OPENSHIFT_VERSIONS` environment variable. It also
+creates a minimal iso image for each version by removing the root filesystem.
+These "template" images are the base for creating the discovery iso.
+
+The image service will take on the responsibility of managing the RHCOS template
+images and creating the minimal ISO template images. This means that the image
+service will require persistent storage (when not using S3).
+
+#### Image Streaming
+
+The version of the image service described here would require editing the image
+while streaming the download rather than the current behavior which stores the
+customized image to be downloaded in a separate call. Allowing the image to be
+stored would complicate the proposal and is discussed in the "Alternatives"
+section.
+
+#### Assisted Service API Changes
+
+For the image service to create the iso it needs some information from the
+assisted service. In the minimal ISO case, it will need the ignition and, if the
+user has configured static networking, an additional ramdisk image. For the full
+ISO, only the ignition is needed.
+
+Today the ignition is created and uploded separately to S3 by the assisted
+service when the discovery image is generated. This proposal involves changing
+that behavior to render the ignition on-demand rather than uploading it every
+time some image information changes. An additional API for the minimal ISO
+ramdisk will behave in the same way.
+
+### Risks and Mitigations
+
+Relying strictly on streaming will increase the load on the image service in
+the SaaS deployment as presigned links directly to S3 would no longer be provided.
+Downloading the image through the service is already required when running
+on a local cluster so this shouldn't be a significant problem, but should
+be load tested. This will be further mitigated by splitting the service as
+many downloads won't affect the other operations of assisted service.
+
+## Design Details [optional]
+
+### Open Questions
+
+- How should the image service API be defined?
+  - A separate swagger file?
+    - Will this require a separate client package?
```

Comment:
I think we should avoid the swagger until complexity requires it.

Reply:
Agree.